### PR TITLE
Team stats

### PR DESCRIPTION
**Pull request by Oxalisviolacea**
This PR:
- completes all of the methods in team stats
- all tests are passing
- still needs to be refactored

### CODE REVIEW

**Comment by ShaunDaneJames** [comment]
Great work team! Thank you for all your hard work.

File: lib/team_stats.rb [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+  end
+
+  def group_teams_data
+    @teams_data.group_by do |row|
+      row[:team_id]
+    end
+  end
+
+  def team_info(team_id)
+    v = []
```
Comment (carson-jardine) [inline lib/team_stats.rb:19]: we should probably use a different variable name here

File: lib/team_stats.rb [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+      if game[:home_team_id] == team_id.to_i
+        game[:away_team_id]
+      else
+        game[:home_team_id]
+      end
+    end
+  end
+
+  def hash_by_opponent_id(team_id)
+    hash = {}
```
Comment (carson-jardine) [inline lib/team_stats.rb:114]: Again here for better variable naming

File: lib/team_stats.rb [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+    opponent_id = favorite_opponent_id(team_id)
+    opponent_name = @teams_data.find do |team|
+      team[:teamname] if opponent_id == team[:team_id]
+    end
+    opponent_name[:teamname]
+  end
+
+  def rival_opponent_id(team_id)
+    rival_opponent = find_percent_of_winning_games_against_rival(team_id).min_by do |rival_id, wins|
+      wins
```
Comment (carson-jardine) [inline lib/team_stats.rb:181]: could prob make this a one liner

**Review by kevxo — APPROVED** [review]
Looks good, but we can try to figure out how to refactor this to be not past 150 lines.

**Review by carson-jardine — APPROVED** [review]

**Comment by carson-jardine** [comment]
Looks great minus some variables needing some renaming :)

**Pull request merged**